server/src/test/java/org/elasticsearch/search/TelemetryMetrics/CoordinatorSearchPhaseAPMMetricsTests.java

```diff
@@ -0,0 +1,149 @@
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertSearchHitsWithoutFailures;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasSize;
+
+public class CoordinatorSearchPhaseAPMMetricsTests extends ESSingleNodeTestCase {
+    private static final String indexName = "test_coordinator_search_phase_metrics";
+    private final int num_primaries = randomIntBetween(2, 7);
+
+    @Override
+    protected boolean resetNodeAfterTest() {
+        return true;
+    }
+
+    @Before
+    private void setUpIndex() throws Exception {
+        createIndex(
+            indexName,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, num_primaries)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                .build()
+        );
+        ensureGreen(indexName);
+
+        prepareIndex(indexName).setId("1").setSource("body", "doc1").setRefreshPolicy(IMMEDIATE).get();
+        prepareIndex(indexName).setId("2").setSource("body", "doc2").setRefreshPolicy(IMMEDIATE).get();
+
+        prepareIndex(CoordinatorSearchPhaseAPMMetricsTests.TestSystemIndexPlugin.INDEX_NAME).setId("1")
+            .setSource("body", "doc1")
+            .setRefreshPolicy(IMMEDIATE)
+            .get();
+        prepareIndex(CoordinatorSearchPhaseAPMMetricsTests.TestSystemIndexPlugin.INDEX_NAME).setId("2")
+            .setSource("body", "doc2")
+            .setRefreshPolicy(IMMEDIATE)
+            .get();
+    }
+
+    @After
+    private void afterTest() {
+        resetMeter();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> getPlugins() {
+        return pluginList(TestTelemetryPlugin.class, CoordinatorSearchPhaseAPMMetricsTests.TestSystemIndexPlugin.class);
+    }
+
+    public void testSearchQueryThenFetch() throws InterruptedException {
+        assertSearchHitsWithoutFailures(
+            client().prepareSearch(indexName).setSearchType(SearchType.QUERY_THEN_FETCH).setQuery(simpleQueryStringQuery("doc1")),
+            "1"
+        );
+        assertMeasurements(List.of(QUERY_SEARCH_PHASE_METRIC));
+    }
+
+    private void resetMeter() {
+        getTestTelemetryPlugin().resetMeter();
+    }
+
+    private TestTelemetryPlugin getTestTelemetryPlugin() {
+        return getInstanceFromNode(PluginsService.class).filterPlugins(TestTelemetryPlugin.class).toList().get(0);
+    }
+
+    private void assertMeasurements(Collection<String> metricNames) {
+        for (var metricName : metricNames) {
+            List<Measurement> measurements = getTestTelemetryPlugin().getLongHistogramMeasurement(metricName);
+            assertThat(measurements, hasSize(1));
+            assertThat(measurements.getFirst().getLong(), greaterThanOrEqualTo(0L));
+        }
+    }
+
+    public static class TestSystemIndexPlugin extends Plugin implements SystemIndexPlugin {
```

Comment:
I'd add some description here about what this plugin provides. It may be the case that in the future somebody would need the same thing and will be able to reuse it.